Sources/WalletConnectUtils/RPCHistory.swift

```diff
@@ -0,0 +1,67 @@
+import JSONRPC
+
+public final class RPCHistory {
+
+    public struct Record: Codable {
+        public enum Origin: String, Codable {
+            case local
+            case remote
+        }
+        let id: RPCID
+        let topic: String
+        let origin: Origin
```

Comment:
just noticed it. What is the purpose of origin?

Reply:
Being able to track whether a request was sent from the client or from the peer. Thinking on long term support, I think is a nice thing to have, might be useful in the future